src/append/rolling_file/rotation.rs

```diff
@@ -31,71 +32,79 @@ pub enum Rotation {
 }
 
 impl Rotation {
-    pub fn next_date_timestamp(&self, current_date: &OffsetDateTime) -> Option<usize> {
+    pub fn next_date_timestamp(&self, current_date: &Timestamp) -> Option<usize> {
+        let timestamp_round = TimestampRound::new().mode(RoundMode::Trunc);
+
         let next_date = match *self {
-            Rotation::Minutely => *current_date + Duration::minutes(1),
-            Rotation::Hourly => *current_date + Duration::hours(1),
-            Rotation::Daily => *current_date + Duration::days(1),
             Rotation::Never => return None,
-        };
-
-        Some(self.round_date(&next_date).unix_timestamp() as usize)
-    }
-
-    fn round_date(&self, date: &OffsetDateTime) -> OffsetDateTime {
-        match *self {
             Rotation::Minutely => {
-                let time = Time::from_hms(date.hour(), date.minute(), 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
+                (*current_date + 1.minute()).round(timestamp_round.smallest(Unit::Minute))
             }
             Rotation::Hourly => {
-                let time = Time::from_hms(date.hour(), 0, 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
-            }
-            Rotation::Daily => {
-                let time = Time::from_hms(0, 0, 0)
-                    .expect("invalid time; this is a bug in logforth rolling file appender");
-                date.replace_time(time)
+                (*current_date + 1.hour()).round(timestamp_round.smallest(Unit::Hour))
             }
-            Rotation::Never => unreachable!("Rotation::Never is impossible to round."),
+            Rotation::Daily => (*current_date + 1.day()).round(timestamp_round.smallest(Unit::Day)),
```

Comment:
How to workaround https://github.com/BurntSushi/jiff/issues/90?

Reply:
Perhaps we can use `RoundMode::Expand`

Reply:
No. Similar issue: Error { inner: ErrorInner { kind: Adhoc(timestamp rounding does not support days), cause: None } }

Reply:
Maybe use `Zoned` everywhere.

Reply:
The docs suggest using `Zoned` when you need to deal with non-uniform units like days. You can still use UTC with `Zoned`.